src/arch/aarch64/CodeGen.zig

```diff
@@ -527,6 +528,28 @@ fn gen(self: *Self) !void {
             self.ret_mcv = MCValue{ .stack_offset = stack_offset };
         }
 
+        for (self.args) |*arg, arg_index| {
+            // Copy register arguments to the stack
+            switch (arg.*) {
+                .register => |reg| {
+                    // The first AIR instructions of the main body are guaranteed
```

Comment:
Are we sure this will always be the case? @andrewrk?

Reply:
As far as I'm aware, this is the case.

https://github.com/ziglang/zig/blob/f68d3c63df0486c7039732ef442c87fcd7c6fc57/src/Air.zig#L31-L33